server/src/main/java/org/opensearch/snapshots/RestoreService.java

```diff
@@ -678,19 +679,27 @@ private Settings getOverrideSettingsInternal() {
 
                         // We will use whatever replication strategy provided by user or from snapshot metadata unless
                         // cluster is remote store enabled or user have restricted a specific replication type in the
-                        // cluster.
+                        // cluster. If cluster is undergoing remote store migration, replication strategy is strictly SEGMENT type
                         if (RemoteStoreNodeAttribute.isRemoteStoreAttributePresent(clusterService.getSettings()) == true
-                            || clusterSettings.get(IndicesService.CLUSTER_INDEX_RESTRICT_REPLICATION_TYPE_SETTING) == true) {
```

Comment:
this logic might not work well during `remote store -> docrep` migration. so, this code runs in cluster manager node, if during migration from `remote store -> docrep`, user updates setting of one of the non cluster manager node first and if cluster manager still have the remote store attributes this will restore the indices as remote store enabled.

Reply:
we should maybe check `(if mode is STRICT and remote store attribute is present) || (migration to remote is ongoing)`.